src/views/PositionPdfPlot/PositionPdfPlotWidget.tsx

```diff
@@ -230,9 +227,23 @@ export const allocate1d = (N: number, value: number | undefined) => {
     return ret
 }
 
-const colorForValue = (v: number) => {
-    const a = Math.max(0, Math.min(255, Math.floor(v * 255) * 3))
-    return [a, a, 60]
+/**
+ * Given a range of values, generates a function that maps a (possibly undefined)
+ * value in that range into an RGBA color value whose R and G intensities are
+ * in (0, 255) and proportional to the value's position within the range.
+ * The generated function returns black transparent pixels for undefined values.
+ * @param min Lowest value in the data range.
+ * @param max Highest value in the data range.
+ * @returns Convenience function to convert values to proportionally colored pixels.
+ */
+const getColorForValueFn = (min: number, max: number) => {
+    const theScale = 255 / (max - min)
+    return (v: number | undefined) => {
+        if (v === undefined) return [0, 0, 0, 0]
+        const proportion = (v - min) * theScale
+        const intensity = Math.max(0, Math.min(255, 3 * Math.floor(proportion)))
+        return [intensity, intensity, 60, 255]
+    }
```

Comment:
So, previously we had a function that was fed a value which was supposed to be the observed value at that pixel, scaled to its percent value from 0-1 over the range of observed values. That led to a lot of code where it was used to do the scaling.

Instead, I decided to replace it with a closure (returned here) that embeds the data range within it, which makes the code above more straightforward. We can also handle the special case of an undefined value here, rather than having to do it above.

Theoretically, we could do this in a fully vectorized way by taking the incoming array, flattening it, transposing the result, and applying a scale-and-offset transform matrix all at one go. There are two tricks with this. First, in order to use the library matrix code, we'd have to do a first pass through the incoming data to make sure we replaced any `undefined` values with 0 (or more properly, with `data_min`, since we need to subtract that out later in scaling). Second, we'd need to invert the y-axis for the entries before doing the `getColorForValue` operation, since that operation needs to return 4 elements (the RGBA channels) for each one position in the data source, and that's not going to play nice.

Between those two hassles, and the need for preprocessing, adding an augmentation row, etc. I figured it wasn't worth actually messing with the matrix math.